crates/structs/src/site.rs

```diff
@@ -130,3 +130,10 @@ pub struct SaveSiteConfig {
   pub config_hjson: String,
   pub auth: String,
 }
+
+#[derive(Serialize)]
+pub struct FederatedInstances {
+  pub linked: Vec<String>,
+  pub allowed: Vec<String>,
+  pub blocked: Vec<String>,
+}
```

Comment:
Alternatively, we could do the check for empty strings on our side, and make these all `Option`s

Reply:
Which check do you mean? I think it makes sense like you have it now, with `federated_instances` being optional.